teps/0123-specifying-on-demand-retry-in-pipelinetask.md

```diff
@@ -0,0 +1,183 @@
+
+The release manager creates a new change request by creating a PR with the details needed for the deployment such as
+an application GitHub repo and a specific branch or commit. Creation of a PR triggers a new `pipelineRun` in which the
+application source is cloned, built, and deployed to the production cluster. After the deployment succeeds, a flaky
+acceptance test is executed against the production deployment. The acceptance tests are flaky and results in failure
+sometimes. This failure prevents updating the change request with the deployment evidences.
+
+Now, the release manager has three choices to work around this flakiness:
+
+1) Configure acceptance-tests task with a reasonable amount of retries. But this might or might not work depending on the
+cause of the flakiness. Retrying a task once it fails might address network connectivity issue as mentioned in the pipeline
+[specification](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-the-retries-field). If the
+flakiness is caused by an issue which requires a fix from the users, this kind of retry does not help.
+
+2) Close the existing PR and create a new PR for the same request. This new PR will trigger a new `pipelineRun` which
+will start from the beginning i.e. clone, build, and deploy the same application source.
+
+3) Create an asynchronous pipeline with just the two tasks `acceptance-test` and `submit-evidence`. Trigger this `pipeline`
+manually with the data (deployment evidence and other configuration needed for the test) from the failed `pipelineRun`.
+
+![CD Use Case](images/0123-cd-use-case.png)
+
+#### CI Use Case
+
+Let's take an example of a CI `pipeline` to validate and test the changes being proposed. Tekton projects has a set of
+checks defined to test the changes from any contributor before merging them upstream.
+
+A contributor creates a PR with the changes and creation of a PR triggers a new `pipelineRun` in which a branch from a
+contributor's forked repo is cloned, the coverage report is generated, a flaky unit test (just like one of our projects)
+and other tests are executed. Now, a flaky unit test often fails and requires a couple of retries before it can
+succeed.
+
+Today, our CI is set up such that we have one `pipeline` per CI job, for example, `unit-test` has its own `pipeline`,
+`integration-test` has its own `pipeline`. With this set up, a flaky unit test can be independently triggered without
+running any other checks.
+
+Let's take this further, in an organization, there is often a single resource to manage CI of multiple applications.
+Now instead of writing multiple CI pipelines for each application, it's easier to manage one common CI pipeline for all
+applications. The CI pipeline is fairly simple and common for all applications. It includes `build`, `test`, and `deploy`.
+Now, when a `unit-test` fails, it's always desirable to just re-run `unit-test` rather than re-running the whole
```

Comment:
Wouldn't this mean that if a user does not detect that the unit test failed and triggered a on-demand retry, the entire pipeline will be kept running and so if you were relying on notifications at the end of a pipelinerun, you wouldn't get it?

Reply:
Yes, it would keep running until timeout.